### PR TITLE
[Enhancement] Adding tests and upgrading packages for ember-cli 0.2

### PR DESCRIPTION
**Pull request by thoov**
Fixes: #2

### CODE REVIEW

**Comment by stefanpenner** [comment]
@thoov lets take a look this afternoon.

File: tests/unit/mixins/buffered-proxy-state-test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    State = BufferedProxy.extend();
+
+    container.register('state:test-state', State, {
+        singleton:   true,
+        instantiate: false
+    });
+
+    Service = Ember.Service.extend(StateMixin, {
+      stateName: 'test-state',
+      container: container
```
Comment (stefanpenner) [inline tests/unit/mixins/buffered-proxy-state-test.js]: we should pass this in to `Service.create` we don't want instance state to leak onto factories.

**Comment by stefanpenner** [comment]
left 1 comment, then it LGTM

**Comment by stefanpenner** [comment]
I added travis-ci

**Comment by jayphelps** [comment]
Will need a rebase due to #12

**Comment by thoov** [comment]
I am closing this and reopening another PR for the tests. Sorry they was open for a few days. I was working on some acceptance tests + a better demo but that is going to take a few more days and I will just send that in another PR.

**Pull request closed without merging**